[DOC] Add policy order to sampling doc (#5694)
grafana/tempo: docs/sources/tempo/set-up-for-tracing/instrument-send/set-up-collector/tail-sampling/policies-strategies.md

| [string_attribute](https://grafana.com/docs/alloy/<ALLOY_VERSION>/reference/components/otelcol/otelcol.processor.tail_sampling/#string_attribute-block)   | Samples based on string attributes (resource and record) value matches.                                        | Filtering specific services or database queries.                        |
| [trace_state](https://grafana.com/docs/alloy/<ALLOY_VERSION>/reference/components/otelcol/otelcol.processor.tail_sampling/#trace_state-block)             | Samples based on TraceState value matches.                                                                     | Implementing complex sampling strategies that rely on trace context.    |

### Policy order

Policies are evaluated in the order they're defined.
When you define your policies, make sure you consider the order in which they're evaluated.
The following points are important to consider:

- Policies are processed sequentially in the order they are defined in the configuration
- Once any policy condition is satisfied, evaluation stops - remaining policies are not checked
- This creates a "first-match wins"

> This is fundamentally about ensuring that your most important observability signals, such as RED (rate, error, duration) metrics, aren't accidentally filtered out by broader sampling rules.

Author: ```suggestion
Being aware of the policy order ensures that your most important observability signals, such as RED (rate, error, duration) metrics, aren't accidentally filtered out by broader sampling rules.
```

For example, add `probabilistic` policies last since they act as a catch-all after all other policies have been evaluated.
If you have three policies in place, `status_code`, `latency`, and `probabilistic`, then the `probabilistic` policy must be the last policy specified so it doesn't prevent the `status_code` and `latency` from being evaluated.
If the `probabilistic` policy samples at 10% and is evaluated first, then 10% of traces were sampled.
No further evaluation on the other policies happens because one of the tail sampling processor conditions has been satisfied.

> This means that traces with errors and high latency are potentially thrown away.

Author: ```suggestion
Traces with errors and high latency are potentially thrown away.
```
